### PR TITLE
DRV-395: Add http/2 support to golang driver

### PR DESCRIPTION
**Pull request by vadimLF**
- Adds golang.org/x/net/http2 module for HTTP/2 support
- Switch to http2.Transport for all requests.

### CODE REVIEW

File: faunadb/client.go [hunk 1]
```diff
@@ -130,8 +134,19 @@ func NewFaunaClient(secret string, configs ...ClientConfig) *FaunaClient {
...
+		if len(client.endpoint) > 5 && client.endpoint[:5] == "https" {
+			dial = nil
+		}
+		transport := &http2.Transport{
+			DialTLS:   dial,
+			AllowHTTP: true,
+		}
+
 		client.http = &http.Client{
-			Timeout: requestTimeout,
```
Comment (erickpintor) [inline faunadb/client.go:134]: Does this mean we're no longer enforcing requests timeout?

Comment (vadimLF) [reply faunadb/client.go:134]: I was thinking about streaming when removed this timeout. I will return timeout for all requests by default.

**Review by erickpintor — APPROVED** [review]
LGTM. I'm happy to see it didn't take too much effort to transition to http2.

**Review by marrony — APPROVED** [review]
lgtm

**Pull request merged**